src/routes/docs/products/auth/custom-token/+page.markdoc

```diff
@@ -0,0 +1,54 @@
+---
+layout: article
+title: Custom token login
+description: Limitless authentication flow in Appwrite. Find out how to implement custom authentication flow or connect to 3rd party authentication providers.
+---
+
+Tokens are short-lived secrets created by server that can be exchanged for session by client. You may already be familiar with tokens if you checked out [Magic URL login](/docs/products/auth/magic-url), [Email OTP login](/docs/products/auth/email-otp) or [Phone (SMS) login](/docs/products/auth/phone-sms).
+
+Custom token allows you to use [Server SDK](/docs/sdks#server) to generate tokens for your own implementations. This allows you to code your own authentication methods using Appwrite Functions or your custom backend. You could implement nick&password sign-in, captcha-protected authentication, phone call auth, and much more. Custom tokens also allow you to skip authentication which is useful when you integrate Appwrite with external authenticaion providers such as Auth0, TypingDNA, or any provider trusted by your users.
+
+# Create custom token {% #init %}
+
+Once you have your server endpoint prepared either in Appwrite Function or custom backend, you can use [Create token](/docs/references/cloud/server-node/users#createToken) endpoint of Users API to generate a token.
+
+```js
+import { Client, Users } from "node-appwrite";
+
+const client = new Client()
+    .setEndpoint('https://cloud.appwrite.io/v1') // Your API Endpoint
+    .setProject('<PROJECT_ID>');               // Your project ID
+    .setKey('<API_KEY>');                    // Your secret API key
+
+const users = new Users(client);
+
+const token = await users.createToken('[USER_ID]');
+
+console.log(token.secret);
+```
+
+Token that got created includes a secret which is 6 character long hexadecimal string. You can configure length of the secret and expiry when creating a token.
+
+If you are integrating with external provider or implementented your own, make sure to validate user authenticated properly before generating a token for him.
+
+If you are implementing token-based authentication flow, share the token secret with user over any channel of your choice instead of directly giving it to him in the response.
```

Comment:
Hmm this is actually pretty complex topic, you need combination of information to do correct decision. Characters used, rate limits, length, expiry, maybe more..

Also we shouldn't be suggesting anything probably, because if FBI wanted to DDoS, they surely can and there is barely anything you can do. Security isn't about solving it or not, it's a balance between convenience and protection. Such a balance is different for every project.

With that said, there is minimum token length of 6 characters (max 128) and minimum expiry of 60 seconds (max 1 year). API is protective enough to not make a silly decision.

Reply:
Okay, as long as we don't endup accidentally encouraging really bad, preventable behaviors